scripts/increment_auto_off.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# A program that ticks a counter down until deactivating log_persistence
+
+import configparser
+import subprocess
+import logging
+import sys
+import json
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+sh = logging.StreamHandler(sys.stdout)
+logger.addHandler(sh)
+
+journalconf = configparser.ConfigParser(strict=False, allow_no_value=True)
+journalconf.read('/etc/systemd/journald.conf')
+
+
+logconf = configparser.ConfigParser(strict=False, allow_no_value=True)
+logconf.read('/var/log/logging.ini')
+
+auto_off_delay = int(logconf.get("logging", "auto_off_delay", fallback="14"))  # Get auto_off_delay, in days. if line is not found/is empty, set to the default 14 days
+persist_state = journalconf.get("Journal", "Storage") == "persistent"
+
+if persist_state and auto_off_delay > 0:
+  body: dict
+  if auto_off_delay >= 1:
+    delay = auto_off_delay - 1
+    body = {
+      "persist_logs": True,
+      "auto_off_delay": delay,
+    }
+    logging.info(f"Persist logs will be automatically turned off in {delay} day(s)")
+  else:
+    body = {
+      "persist_logs": False,
+      "auto_off_delay": 14,  # Reset to default for next time
+    }
+    logging.info("Persist logs has been turned off automatically")
+
+  json_data = json.dumps(body)
+  subprocess.run([
+      'curl', '-X', 'POST',
+      '--header', 'Content-Type: application/json',
+      '--data', json_data,
+      'http://localhost:5001/settings/persist_logs',
+  ], check=True)
```

Comment:
Everything in through the front door, no editing the ini files directly from this script

Reply:
it looks like you get the status of the persistance timing from the config file, but set anything new via the web interface. it feels like you should pick one interface and stick with it, imo

Reply:
also, i'd request that you don't shell out to curl to set this. use `requests`, or if you don't want something outside of the standard lib use `urllib.request`.